### DISCUSSION TITLE
Turning on/off only certain ewmhrequests

### DISCUSSION THREAD

**Discussion by patrick96** [herbstluftwm/herbstluftwm#1488] [Q&A] [answered]
Turning off `ewmhrequests` for a window often allows me to put the application in fullscreen mode without the window itself taking up the whole screen (windowed fullscreen). However, it also turns off "demands attention" requests from the window.

Is it possible to turn off only the fullscreen requests while leaving others (including "demands attention") turned on?

**Comment by t-wissmann** [comment]
Actually `ewmhnotify` should be responsible for preventing the window from knowing that it is fullscreen (and not `ewmhrequests`). But moreover, it might be enough to disable `ewmhnotify` when changing the fullscreen state, so the following keybinding should accomplish "windowed fullscreen":
```
hc keybind Mod4-Control-f chain , attr clients.focus.ewmhnotify toggle , fullscreen toggle , attr clients.focus.ewmhnotify toggle
```
Many people who ask for windows fullscreen actually want window maximization. If so, you can take a loot at `maximize.sh` that comes with hlwm.

**Reply by patrick96** [reply]
Hi, thanks for the reply :)

I didn't know about `maximize.sh`, looks pretty cool, I always just put windows in fullscreen ;)

But that's not what I was intending to ask in my original question, but I think I used "windowed fullscreen" wrong. What I want is to activate a program's fullscreen mode (e.g. making a youtube video take up the whole window) while the window itself stays the same. I can achieve this by disabling `ewmhrequests`, however this also prevents the window from signaling "demands attention".

Is there a way to only turn off `ewmhrequests` for the fullscreen request but keep it turned on for signaling urgency?

**Comment by t-wissmann** [comment, marked as answer]
For faking the fullscreen state to the client, try this:
```
hc keybind Mod4-Control-f chain , attr clients.focus.ewmhrequests off , attr clients.focus.ewmhnotify on , attr clients.focus.fullscreen on , attr clients.focus.ewmhnotify off , set_attr clients.focus.fullscreen off , set_attr clients.focus.ewmhnotify on , sprintf CMD 'sleep 2 ; herbstclient attr clients.%s.ewmhrequests on' clients.focus.winid spawn bash -c CMD
```
It's a bit of a hack, I'm enabling 'ewmhrequests' on purpose a bit later because e.g. firefox directly sends some requests when hlwm changes its fullscreen state.

**Reply by LucasTavaresA** [reply]
this would be really nice as a rule

**Reply by patrick96** [reply]
Clever hack ;)

This only works if the window actually properly reacts to the WM putting it in fullscreen mode. But I guess you could also just trigger the windows own fullscreen mode during the 2 second wait.
